This patch is to prevent overwrite of priority set within switch.

--- a/net/8021q/vlan_core.c
+++ b/net/8021q/vlan_core.c
@@ -128,7 +128,9 @@ bool vlan_do_receive(struct sk_buff **sk
 		skb_reset_mac_len(skb);
 	}
 
+#ifndef CONFIG_LANTIQ_IPQOS
 	skb->priority = vlan_get_ingress_priority(vlan_dev, skb->vlan_tci);
+#endif
 
 #ifdef CONFIG_TI_PACKET_PROCESSOR
     /* 
--- a/net/ipv4/ip_forward.c
+++ b/net/ipv4/ip_forward.c
@@ -178,7 +178,9 @@ int ip_forward(struct sk_buff *skb)
 	if (rt->rt_flags&RTCF_DOREDIRECT && !opt->srr && !skb_sec_path(skb))
 		ip_rt_send_redirect(skb);
 
+#ifndef CONFIG_LANTIQ_IPQOS
 	skb->priority = rt_tos2priority(iph->tos);
+#endif
 
 	return NF_HOOK(NFPROTO_IPV4, NF_INET_FORWARD, skb, skb->dev,
 		       rt->dst.dev, ip_forward_finish);
